Task3/app/src/main/java/com/example/task3/MainActivity.kt

```diff
@@ -0,0 +1,97 @@
+package com.example.task3
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.enableEdgeToEdge
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import com.example.task3.ui.theme.Task3Theme
+import android.content.BroadcastReceiver
+import android.content.ComponentName
+import android.widget.TextView
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.ServiceConnection
+import android.os.IBinder
+import android.os.RemoteException
+import com.example.task3.ISystemUptimeService
+import android.util.Log
+
+
+
+class MainActivity : ComponentActivity() {
+
+    private var uptimeService : ISystemUptimeService? = null
+    private var isBound = false
+
+    /* Service Connection */
+    private val connection = object : ServiceConnection {
+
+        override fun onServiceConnected(name: ComponentName?, service: IBinder?) {
+            uptimeService = ISystemUptimeService.Stub.asInterface(service)
+            isBound = true
+            updateUptime()
+        }
+
+        override fun onServiceDisconnected(name: ComponentName?) {
+            uptimeService = null
```

Comment:
what is the purpose of `isBound` when `uptimeService` will be null if the service is not connected?

Reply:
Correct, this flag is redundant  "uptimeService" variable must be enough to check connection status. I removed the "isBound" and updated the code to use the stat of "uptimeService"